#270 Fix cross page links with anchors are rendered incorrectly

agrz opened the pull request:
This MR fixes #270 
Cross references with or without anchors are now rendered correctly and anchors are correctly navigated.
Tested cross-page references using different formats and all work correct with the changes:
 <<page.adoc#,reference>> 
 <<page.adoc#_anchor,reference>> 
xref:page.adoc[reference]
xref:page.adoc#_anchor[reference]

Internal Cross references on same page also keep working:
<<_anchor,reference>>

Tested with some quite complex page tree with references we used internally (Tested on Confluence 7.16.5).

cstettler commented:
Hi @agrz, thank you very much for your valuable pull request. 

The documentation currently mentions that references to other pages using anchors is not yet supported (see https://confluence-publisher.atlassian.net/wiki/spaces/CPD/pages/315916297/Pages at the bottom).

With your contribution, this is no longer the case. Would you mind removing the note in the documentation and instead provide an example (similar to "Links between Pages", e.g. "Links between Pages with Anchors") that demonstrates the support you are about to add. That would be highly appreciated, thank you!

agrz commented:
Hi Christian,

of course. I did not realize there was this warning in the documentation. I updated the pages documentation and removed the warning.

Best regards

Alexander

cstettler commented:
Thank you very much @agrz for the documentation update!

cstettler commented:
@agrz, Confluence Publisher version `0.19.0` containing your contribution is now available on Maven Central and Docker Hub.
